collective/googlenews/controlpanel.py

```diff
@@ -1,22 +1,87 @@
 # -*- coding: utf-8 -*-
-
-from plone.app.registry.browser import controlpanel as base
-
 from collective.googlenews import _
+from collective.googlenews.interfaces import GoogleNewsEditorFieldsGroup
 from collective.googlenews.interfaces import GoogleNewsSettings
+from collective.googlenews.interfaces import IGoogleNewsSchema
+from plone.app.registry.browser import controlpanel as base
+from plone.formwidget.namedfile.widget import NamedImageFieldWidget
+from plone.registry.interfaces import IRegistry
+from zope.component._api import getUtility
+from zope.interface import alsoProvides
 
 
 class SettingsEditForm(base.RegistryEditForm):
 
-    schema = GoogleNewsSettings
+    """Google news control panel settings form."""
+
+    schema = IGoogleNewsSchema
     label = _(u"Google News Settings")
     description = _(u"controlpanel_desc",
                     default=u"You can configure the settings of collective.googlenews add-on")
 
+    fields = GoogleNewsSettings
+    groups = (GoogleNewsEditorFieldsGroup,)
+
+    def getContent(self):
+        """Set multiple registry schema."""
+        return AbstractRecordsProxy(self.schema)
+
     def updateWidgets(self):
+        """Fix portal_types widget width."""
         super(SettingsEditForm, self).updateWidgets()
         self.widgets['portal_types'].style = u'min-width: 200px;'
 
+    def updateFields(self):
+        """Change google news logo widget."""
+        super(SettingsEditForm, self).updateFields()
+        group = self.groups[0]
+        group.fields['googlenews_logo']\
+            .widgetFactory = NamedImageFieldWidget
+
+
+class AbstractRecordsProxy(object):
```

Comment:
http://stackoverflow.com/questions/33610919/what-is-the-canonical-way-to-create-tabs-to-a-control-panel-configlet